### PR TITLE
Fix filter UI issues

### PR DESCRIPTION
**Pull request by ranquild**
Fix paddings, text overflow, colors in various places.

For color changes, see this https://github.com/metabase/metabase/pull/37443

Screenshot 2024-01-14 at 17 53 26
Screenshot 2024-01-14 at 17 53 38
Screenshot 2024-01-14 at 17 53 46
Screenshot 2024-01-14 at 17 54 40

### CODE REVIEW

File: frontend/src/metabase-lib/types.ts [hunk 1]
```diff
@@ -138,6 +138,8 @@ export type ColumnDisplayInfo = {
   isCalculated: boolean;
   isFromJoin: boolean;
   isImplicitlyJoinable: boolean;
+  isAggregation: boolean;
```
Comment (ranquild) [inline frontend/src/metabase-lib/types.ts:141]: These properties are always returned but we didn't have type for them.

File: frontend/src/metabase/querying/components/FilterValuePicker/utils.ts [hunk 2]
```diff
@@ -30,10 +30,12 @@ export function canSearchFieldValues(
 }
 
 function getFieldOptions(fieldValues: FieldValue[]): SelectItem[] {
-  return fieldValues.map(([value, label = value]) => ({
-    value: String(value),
-    label: String(label),
-  }));
+  return fieldValues
+    .filter(([value]) => value != null)
```
Comment (ranquild) [inline frontend/src/metabase/querying/components/FilterValuePicker/utils.ts:34]: Before:
Screenshot 2024-01-14 at 15 11 31

After:
Screenshot 2024-01-14 at 15 13 24

There was an old comments about having to exclude nulls from field values, but I forgot about it.

File: frontend/src/metabase/ui/components/buttons/ActionIcon/ActionIcon.styled.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import type { MantineThemeOverride } from "@mantine/core";
+
+export const getActionIconOverrides =
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/buttons/ActionIcon/ActionIcon.styled.tsx:3]: Used for close button of the modal, not exposed on its own yet.

File: frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx [hunk 4]
```diff
@@ -34,6 +35,11 @@ export const getButtonOverrides = (): MantineThemeOverride["components"] => ({
         },
         label: {
           ref: getStylesRef("label"),
+          display: "inline-block",
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx:38]: Fixes text overflow with buttons

File: frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx [hunk 5]
```diff
@@ -8,12 +8,6 @@ export const getModalOverrides = (): MantineThemeOverride["components"] => ({
         fontSize: theme.fontSizes.xl,
         fontWeight: 700,
       },
-      close: {
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx:11]: Fixed in ActionIcon properly

File: frontend/src/metabase/querying/components/FilterPicker/FilterOperatorPicker/FilterOperatorPicker.tsx [hunk 6]
```diff
@@ -36,6 +36,7 @@ export function FilterOperatorPicker<T extends string>({
     <Select
       data={data}
       value={value}
+      miw="14rem"
```
Comment (ranquild) [inline frontend/src/metabase/querying/components/FilterPicker/FilterOperatorPicker/FilterOperatorPicker.tsx:39]: Fixes overflow issues for large operators, like "Greater than or equal to"

File: frontend/src/metabase-lib/queries/utils/dataset.js [hunk 7]
```diff
@@ -18,21 +18,6 @@ export function fieldRefForColumn(column) {
   );
 }
 
-/**
- * finds the column object from the dataset results for the given `table.columns` column setting
- * @param  {Column[]} columns             Dataset results columns
- * @param  {ColumnSetting} columnSetting  A "column setting" from the `table.columns` settings
- * @return {?Column}                      A result column
- */
-export function findColumnForColumnSetting(columns, columnSetting) {
```
Comment (ranquild) [inline frontend/src/metabase-lib/queries/utils/dataset.js:27]: Unused

File: frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx [hunk 8]
```diff
@@ -34,6 +35,11 @@ export const getButtonOverrides = (): MantineThemeOverride["components"] => ({
...
+          height: "auto",
```
Comment (kamilmielnik) [inline frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx:39]: What does this do?

File: frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx [hunk 9]
```diff
@@ -34,6 +35,11 @@ export const getButtonOverrides = (): MantineThemeOverride["components"] => ({
...
+          overflow: "hidden",
+          whiteSpace: "nowrap",
```
Comment (kamilmielnik) [inline frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx:41]: These 2 properties are already there

```suggestion
```

**Review by kamilmielnik — APPROVED** [review]

**Review by uladzimirdev — APPROVED** [review]
Kamil's comments regarding styles are still valid

File: frontend/src/metabase/ui/components/buttons/ActionIcon/ActionIcon.styled.tsx (hunk 3 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/ui/components/buttons/ActionIcon/ActionIcon.styled.tsx:3]: @ranquild isn't `Modal` using the `CloseButton`?
https://mantine.dev/core/close-button/

Comment (nemanjaglumac) [reply frontend/src/metabase/ui/components/buttons/ActionIcon/ActionIcon.styled.tsx:3]: https://github.com/metabase/metabase/blob/master/frontend/src/metabase/querying/components/FilterModal/FilterModal.tsx#L105

**Pull request merged**